### PR TITLE
[0236] Vehicle Data TireStatus

### LINKED ISSUE #1089: [SDL 0236] Update mismatch in TireStatus structure
Proposal: [Update mismatch in TireStatus structure](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0236-TireStatus-Mismatch.md)

> This proposal is about correcting a mismatch between HMI_API and MOBILE_API for the `TireStatus` structure.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/743

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the following revision: clarify that this update will be a breaking change.

> The proposal .md file was updated to reflect these revisions on 6/6/19.

### PR DESCRIPTION
**Pull request by kostyaBoss**
Fixes #1089

This PR is **[ready]** for review.

### Related PR's
[Core](https://github.com/smartdevicelink/sdl_core/pull/3583)

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behavior
- [x] I have tested Android, Java SE, and Java EE

### Summary
Changes according to #1089

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by kostyaBoss** [comment]
@santanamk this PR is ready for the Ford review

**Comment by joeygrover** [comment]
@kostyaBoss

1. Please avoid removing sections from the PR template and instead filling them out with detailed and helpful information.
2. You did not check you have signed the CLA.
3. You do not provide a Core branch/PR that this was tested against and therefore this PR is not reviewable at this time.
4. The TireStatus.java file is completely overwritten, this makes a code review very difficult. It is likely you simply ran the RPC generator and committed the results. This is not an acceptable method for the Java Suite library as there are many corner cases in the RPC classes that the generator does not have information on. The generator is used for new classes, however, modification on existing classes should not come from the generator as it will overwrite the file as seen in this PR. The committed changes should only be the code that actually changed.

**Comment by vladmu** [comment]
@joeygrover thank you for your feedback, we appreciate your help and will fix all problems you mentioned. However please note that this PR currently is under internal and Ford review, therefore some problems could be present here. We'll do all our best to minimize them before this PR goes to the PMs review.

**Comment by joeygrover** [comment]
@vladmu I am not performing a content review of the code itself, but as the project maintainer if I see a pull request that is very off it is in the best interest of the SDLC to comment on it right away and ensure that those items can be addressed as to avoid delays. Any pull request in the open will always be visible to the community and will be open to comments, especially those from the project maintainer. We await Ford's approval before performing a thorough review.

**Comment by kostyaBoss** [comment]
@santanamk ready for the Ford review

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/Validator.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/Validator.java:1988]: @kostyaBoss I would suggest reverting this change.  These are valid changes, since the `pressureTellTale` functions are being renamed per the parameter, but it is out of scope of the PR.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/VehicleDataHelper.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/VehicleDataHelper.java:231]: @kostyaBoss I would suggest reverting these changes in `VehicleDataHelper.java`.  These are valid changes, since the `pressureTellTale` functions are being renamed per the parameter, but is out of scope of the PR.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/TireStatusTest.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/TireStatusTest.java:26]: @kostyaBoss I would suggest reverting these changes in `TireStatusTest.java`.  These are valid changes, since the `pressureTellTale` functions are being renamed per the parameter, but is out of scope of the PR.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/notifications/OnVehicleDataTests.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/notifications/OnVehicleDataTests.java:261]: @kostyaBoss I would suggest reverting these changes in `OnVehicleDataTests.java`. These are valid changes, since the `pressureTellTale` functions are being renamed per the parameter, but is out of scope of the PR.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/responses/GetVehicleDataResponseTests.java
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/responses/GetVehicleDataResponseTests.java:142]: @kostyaBoss I would suggest reverting these changes in `GetVehicleDataResponseTests.java`. These are valid changes, since the `pressureTellTale` functions are being renamed per the parameter, but is out of scope of the PR.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java:40]: @kostyaBoss It looks like this whole file `TireStatus.java` was regenerated with the generator tool.  Could you just merge in the changes that are necessary for this PR?

**Review by santhanamk — COMMENTED** [review]
@kostyaBoss I reviewed this PR.  I have made a few suggestions.  Please take a look whenever you get the chance.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java
Comment (kostyaBoss) [reply base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java:40]: @santhanamk double-checked the TireStatus, there are only documentation updates, should I still revert?

**Comment by kostyaBoss** [comment]
@santhanamk I've processed the PR comments, please re-review

**Review by santhanamk — COMMENTED** [review]
> @santhanamk I've processed the PR comments, please re-review
> 
> After the updates rollback, I can see constantly failing tests with message
> 
> `junit.framework.AssertionFailedError: Method: getPressureTelltale cannot be found for RPC: TireStatus. Make sure that the method exists and that the parameters order and types are identical to the RPC specs at junit.framework.Assert.fail(Assert.java:50) `
> 
> I think that happens because the param naming mismatches one in the XML file. LMK If I should rollback changes which are causing this issue

@kostyaBoss What steps are you taking to recreate this error?  Using your code, all of the unit tests passed and I don't see this error.

**Review by santhanamk — APPROVED** [review]
@kostyaBoss Code looks good. I have approved the PR.

Once you have tested against Core and HMI (if applicable), please put the links for those PRs. Please mark the PR as ready for review, and you can tag Livio once the testing is done on your end.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/TireStatus.java:185]: @kostyaBoss I would delete your `setPressureTellTale` and `getPressureTellTale` functions and replace it with what was there before:

```
/**
     * @deprecated use {@link #setPressureTelltale(WarningLightStatus pressureTellTale)} instead.
     */
    @Deprecated
    public TireStatus setPressureTellTale(@NonNull WarningLightStatus pressureTellTale) {
        return setPressureTelltale(pressureTellTale);
    }

    /**
     * @deprecated use {@link #getPressureTelltale()} instead.
     */
    @Deprecated
    public WarningLightStatus getPressureTellTale() {
        return getPressureTelltale();
    }

    /**
     * Sets the status of the tire pressure Telltale.
     *
     * @param pressureTellTale the status of the tire pressure Telltale.
     */
    public TireStatus setPressureTelltale(@NonNull WarningLightStatus pressureTellTale) {
        setValue(KEY_PRESSURE_TELL_TALE, pressureTellTale);
        return this;
    }

    /**
     * Gets the status of the tire pressure Telltale.
     *
     * @return the status of the tire pressure Telltale.
     */
    public WarningLightStatus getPressureTelltale() {
        return (WarningLightStatus) getObject(WarningLightStatus.class, KEY_PRESSURE_TELL_TALE);
    }
```

Once you do this, you shouldn't see the errors in the `RPCGenericTests`.  Please try this and let me know.

**Review by santhanamk — COMMENTED** [review]
@kostyaBoss Please try out the suggested change in `TireStatus.java` and see if that resolves the issue you are seeing with the `RPCGenericTests`

**Comment by kostyaBoss** [comment]
@santhanamk I've processed the changes, please re-review

**Review by santhanamk — APPROVED** [review]
@kostyaBoss Code looks good.  I have approved this PR.

**Comment by kostyaBoss** [comment]
@bilal-alsharifi , @jordynmackool the PR is ready for the review

**Pull request closed without merging**
